granulate_utils/linux/ns.py

```diff
@@ -59,19 +59,23 @@ def resolve_proc_root_links(proc_root: str, ns_path: str) -> str:
     parts = Path(ns_path).parts
 
     path = proc_root
+    seen = set()
     for part in parts[1:]:  # skip the / (or multiple /// as .parts gives them)
         next_path = os.path.join(path, part)
-        if os.path.islink(next_path):
+        while os.path.islink(next_path):
+            if next_path in seen:
+                raise OSError("Symlink loop detected")
+            seen.add(next_path)
             link = os.readlink(next_path)
             if os.path.isabs(link):
                 # absolute - prefix with proc_root
                 next_path = proc_root + link
             else:
                 # relative: just join
-                next_path = os.path.join(path, link)
+                next_path = os.path.join(os.path.dirname(next_path), link)
         path = next_path
 
-    return path
+    return os.path.abspath(path)
```

Comment:
Why do you need `abspath` here? This function doesn't respect the concept of `/proc/pid/root`.

Reply:
This was to resolve `..` in the path, but if we don't care about that I can remove it

Reply:
Yup, can remove